pkg/fixedpoint/convert.go

```diff
@@ -115,9 +115,50 @@ func (v Value) FormatString(prec int) string {
 	} else if v == NegInf {
 		return "-inf"
 	}
-	pow := math.Pow10(prec)
-	return strconv.FormatFloat(
-		math.Trunc(float64(v)/DefaultPow*pow)/pow, 'f', prec, 64)
+
+	u := int64(v)
+
+	// trunc precision
+	precDiff := DefaultPrecision - prec
+	if precDiff > 0 {
```

Comment:
what about negative precDiff?

Reply:
precDiff > 0 means the prec is larger than 8 and our precision is only 8 for fixedpoint.Value
As a result, we don't need to trunc it.

Reply:
if prec = 5, precDiff = 8 - 5 = 3.
For 123456789 (1.23456789) => 123456000 (1.23456)